packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php

```diff
@@ -0,0 +1,173 @@
+use Webkul\Sales\Models\OrderAddress as OrderAddressModel;
+use Webkul\Sales\Models\OrderItem as OrderItemModel;
+use Webkul\Sales\Models\OrderPayment as OrderPaymentModel;
+use Webkul\Sales\Models\OrderTransaction;
+
+use function Pest\Laravel\get;
+use function Pest\Laravel\postJson;
+
+afterEach(function () {
+    // Cleaning up the row  which are creating
+    CustomerModel::query()->delete();
+    OrderAddressModel::query()->delete();
+    OrderModel::query()->delete();
+    OrderPaymentModel::query()->delete();
+    CartItemModel::query()->delete();
+    CartModel::query()->delete();
+    ProductModel::query()->delete();
+    InvoiceModel::query()->delete();
+});
+it('should return the index page of transactions', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    get(route('admin.sales.transactions.index'))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.sales.transactions.index.title'));
+});
+
+it('should store the transaction into the database', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    // Arrange
+    $product = (new ProductFaker([
+        'attributes' => [
+            5 => 'new',
+        ],
+
+        'attribute_value' => [
+            'new' => [
+                'boolean_value' => true,
+            ],
+        ],
+    ]))
+        ->getSimpleProductFactory()
+        ->create();
+
+    $order = OrderModel::factory()->create([
+        'cart_id' => $cartId = CartItemModel::factory()->create([
+            'product_id' => $product->id,
+        ])->id,
+    ]);
+
+    OrderItemModel::factory()->create([
+        'product_id' => $product->id,
+        'order_id'   => $order->id,
+    ]);
+
+    $payment = OrderPaymentModel::factory()->create([
+        'order_id' => $order->id,
+    ]);
+
+    OrderAddressModel::factory()->create([
+        'order_id'     => $order->id,
+        'cart_id'      => $cartId,
+        'address_type' => OrderAddressModel::ADDRESS_TYPE_BILLING,
+    ]);
+
+    $invoice = InvoiceModel::factory([
+        'order_id'         => $order->id,
+        'sub_total'        => $order->grand_total,
+        'base_sub_total'   => $order->grand_total,
+        'grand_total'      => $order->grand_total,
+        'base_grand_total' => $order->grand_total,
+        'increment_id'     => app(InvoiceSequencer::class)->resolveGeneratorClass(),
+    ])->create();
+
+    postJson(route('admin.sales.transactions.store'), [
+        'invoice_id'     => $invoice->id,
+        'payment_method' => $payment->method,
+        'amount'         => $order->grand_total,
+    ])
+        ->assertRedirect(route('admin.sales.transactions.index'))
+        ->isRedirection();
+
+    $this->assertDatabaseHas('order_transactions', [
+        'status'     => 'paid',
+        'invoice_id' => $invoice->id,
+        'order_id'   => $order->id,
+    ]);
+});
+
+it('should view the transaction', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    // Arrange
```

Comment:
All arrangements should be done first. It should follow AAA concept.

Arrange, Act & Assert

Reply:
done